File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/NativeViewHierarchyManager.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.uimanager;

import android.view.View;
import androidx.annotation.Nullable;
import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.common.annotations.internal.LegacyArchitecture;
import com.facebook.react.common.annotations.internal.LegacyArchitectureLogLevel;
import com.facebook.react.common.annotations.internal.LegacyArchitectureLogger;

/**
 * @deprecated This class is part of Legacy Architecture and has been stubbed out. It will be
 *     removed in a future release.
 */
@LegacyArchitecture(logLevel = LegacyArchitectureLogLevel.ERROR)
@Deprecated(
    since = "This class is part of Legacy Architecture and will be removed in a future release")
public class NativeViewHierarchyManager {

  static {
    LegacyArchitectureLogger.assertLegacyArchitecture(
        "NativeViewHierarchyManager", LegacyArchitectureLogLevel.ERROR);
  }

  /**
   * @deprecated Use new architecture instead.
   */
  public NativeViewHierarchyManager(ViewManagerRegistry viewManagers) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public NativeViewHierarchyManager(ViewManagerRegistry viewManagers, RootViewManager manager) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public final synchronized @Nullable View resolveView(int tag) {
    return null;
  }

  /**
   * @deprecated Use new architecture instead.
   */
  public final synchronized @Nullable ViewManager resolveViewManager(int tag) {
    return null;
  }

  /**
   * @deprecated Use new architecture instead.
   */
  public void setLayoutAnimationEnabled(boolean enabled) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void updateInstanceHandle(int tag, long instanceHandle) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void updateProperties(int tag, ReactStylesDiffMap props) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void updateViewExtraData(int tag, Object extraData) {}

  /**
   * @deprecated Please use {@link #updateLayout(int tag, int x, int y, int width, int height,
   *     YogaDirection layoutDirection)} instead.
   */
  @Deprecated
  public void updateLayout(int tag, int x, int y, int width, int height) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void updateLayout(
      int parentTag,
      int tag,
      int x,
      int y,
      int width,
      int height,
      com.facebook.yoga.YogaDirection layoutDirection) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized long getInstanceHandle(int reactTag) {
    return 0;
  }

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void createView(
      ThemedReactContext themedContext,
      int tag,
      String className,
      @Nullable ReactStylesDiffMap initialProps) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void manageChildren(
      int tag,
      @Nullable int[] indicesToRemove,
      @Nullable ViewAtIndex[] viewsToAdd,
      @Nullable int[] tagsToDelete) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void setChildren(int tag, ReadableArray childrenTags) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void addRootView(int tag, View view) {}

  /**
   * @deprecated Use new architecture instead.
   */
  protected final synchronized void addRootViewGroup(int tag, View view) {}

  /**
   * @deprecated Use new architecture instead.
   */
  protected synchronized void dropView(View view) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void removeRootView(int rootViewTag) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized int getRootViewNum() {
    return 0;
  }

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void measure(int tag, int[] outputBuffer) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void measureInWindow(int tag, int[] outputBuffer) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized int findTargetTagForTouch(int reactTag, float touchX, float touchY) {
    return 0;
  }

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void setJSResponder(
      int reactTag, int initialReactTag, boolean blockNativeResponder) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void clearJSResponder() {}

  /**
   * @deprecated Use new architecture instead.
   */
  @Deprecated
  public synchronized void dispatchCommand(
      int reactTag, int commandId, @Nullable ReadableArray args) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void dispatchCommand(
      int reactTag, String commandId, @Nullable ReadableArray args) {}

  /**
   * @deprecated Use new architecture instead.
   */
  public synchronized void sendAccessibilityEvent(int tag, int eventType) {}
}

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/NativeViewHierarchyManager.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.uimanager

import android.view.View
import com.facebook.react.bridge.ReadableArray
import com.facebook.react.common.annotations.internal.LegacyArchitecture
import com.facebook.react.common.annotations.internal.LegacyArchitectureLogLevel
import com.facebook.react.common.annotations.internal.LegacyArchitectureLogger

/**
 * This class is part of Legacy Architecture and has been stubbed out. It will be removed in a
 * future release.
 */
@LegacyArchitecture(logLevel = LegacyArchitectureLogLevel.ERROR)
@Deprecated("This class is part of Legacy Architecture and will be removed in a future release")
@Suppress("DEPRECATION")
public open class NativeViewHierarchyManager {

  @Deprecated("Use new architecture instead.") public constructor(viewManagers: ViewManagerRegistry)

  @Deprecated("Use new architecture instead.")
  internal constructor(viewManagers: ViewManagerRegistry, manager: RootViewManager)

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public fun resolveView(tag: Int): View? = null

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public fun resolveViewManager(tag: Int): ViewManager<*, *>? = null

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  public open fun setLayoutAnimationEnabled(enabled: Boolean): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun updateInstanceHandle(tag: Int, instanceHandle: Long): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun updateProperties(tag: Int, props: ReactStylesDiffMap?): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun updateViewExtraData(tag: Int, extraData: Any?): Unit = Unit

  /** @deprecated Please use [updateLayout] with YogaDirection parameter instead. */
  @Deprecated("Please use updateLayout with YogaDirection parameter instead.")
  public open fun updateLayout(tag: Int, x: Int, y: Int, width: Int, height: Int): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun updateLayout(
      parentTag: Int,
      tag: Int,
      x: Int,
      y: Int,
      width: Int,
      height: Int,
      layoutDirection: com.facebook.yoga.YogaDirection,
  ): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun getInstanceHandle(reactTag: Int): Long = 0

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun createView(
      themedContext: ThemedReactContext,
      tag: Int,
      className: String,
      initialProps: ReactStylesDiffMap?,
  ): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  internal open fun manageChildren(
      tag: Int,
      indicesToRemove: IntArray?,
      viewsToAdd: Array<ViewAtIndex>?,
      tagsToDelete: IntArray?,
  ): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun setChildren(tag: Int, childrenTags: ReadableArray?): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun addRootView(tag: Int, view: View): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  protected fun addRootViewGroup(tag: Int, view: View): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  protected open fun dropView(view: View): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun removeRootView(rootViewTag: Int): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun getRootViewNum(): Int = 0

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun measure(tag: Int, outputBuffer: IntArray): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun measureInWindow(tag: Int, outputBuffer: IntArray): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun findTargetTagForTouch(reactTag: Int, touchX: Float, touchY: Float): Int = 0

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun setJSResponder(
      reactTag: Int,
      initialReactTag: Int,
      blockNativeResponder: Boolean,
  ): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun clearJSResponder(): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun dispatchCommand(reactTag: Int, commandId: Int, args: ReadableArray?): Unit = Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun dispatchCommand(reactTag: Int, commandId: String, args: ReadableArray?): Unit =
      Unit

  /** @deprecated Use new architecture instead. */
  @Deprecated("Use new architecture instead.")
  @Synchronized
  public open fun sendAccessibilityEvent(tag: Int, eventType: Int): Unit = Unit

  public companion object {
    init {
      LegacyArchitectureLogger.assertLegacyArchitecture(
          "NativeViewHierarchyManager",
          LegacyArchitectureLogLevel.ERROR,
      )
    }
  }
}
